xbmc/utils/LegacyPathTranslation.cpp

```diff
@@ -11,95 +11,97 @@
 #include "URL.h"
 #include "utils/StringUtils.h"
 
-typedef struct Translator {
-  const char *legacyPath;
-  const char *newPath;
-} Translator;
+#include <string>
+#include <unordered_map>
+
+namespace
+{
+
+using namespace std::string_literals;
 
 // ATTENTION: Make sure the longer match strings go first
 // because the string match is performed with StringUtils::StartsWith()
-static Translator s_videoDbTranslator[] = {
-  { "videodb://1/1",  "videodb://movies/genres" },
-  { "videodb://1/2",  "videodb://movies/titles" },
-  { "videodb://1/3",  "videodb://movies/years" },
-  { "videodb://1/4",  "videodb://movies/actors" },
-  { "videodb://1/5",  "videodb://movies/directors" },
-  { "videodb://1/6",  "videodb://movies/studios" },
-  { "videodb://1/7",  "videodb://movies/sets" },
-  { "videodb://1/8",  "videodb://movies/countries" },
-  { "videodb://1/9",  "videodb://movies/tags" },
-  { "videodb://1",    "videodb://movies" },
-  { "videodb://2/1",  "videodb://tvshows/genres" },
-  { "videodb://2/2",  "videodb://tvshows/titles" },
-  { "videodb://2/3",  "videodb://tvshows/years" },
-  { "videodb://2/4",  "videodb://tvshows/actors" },
-  { "videodb://2/5",  "videodb://tvshows/studios" },
-  { "videodb://2/9",  "videodb://tvshows/tags" },
-  { "videodb://2",    "videodb://tvshows" },
-  { "videodb://3/1",  "videodb://musicvideos/genres" },
-  { "videodb://3/2",  "videodb://musicvideos/titles" },
-  { "videodb://3/3",  "videodb://musicvideos/years" },
-  { "videodb://3/4",  "videodb://musicvideos/artists" },
-  { "videodb://3/5",  "videodb://musicvideos/albums" },
-  { "videodb://3/9",  "videodb://musicvideos/tags" },
-  { "videodb://3",    "videodb://musicvideos" },
-  { "videodb://4",    "videodb://recentlyaddedmovies" },
-  { "videodb://5",    "videodb://recentlyaddedepisodes" },
-  { "videodb://6",    "videodb://recentlyaddedmusicvideos" }
+const auto s_videoDbTranslator = std::unordered_map<std::string,std::string> {
+  { "videodb://1/1"s, "videodb://movies/genres"s },
+  { "videodb://1/2"s, "videodb://movies/titles"s },
+  { "videodb://1/3"s, "videodb://movies/years"s },
+  { "videodb://1/4"s, "videodb://movies/actors"s },
+  { "videodb://1/5"s,  "videodb://movies/directors"s },
+  { "videodb://1/6"s,  "videodb://movies/studios"s },
+  { "videodb://1/7"s,  "videodb://movies/sets"s },
+  { "videodb://1/8"s,  "videodb://movies/countries"s },
+  { "videodb://1/9"s,  "videodb://movies/tags"s },
+  { "videodb://1"s,    "videodb://movies"s },
+  { "videodb://2/1"s,  "videodb://tvshows/genres"s },
+  { "videodb://2/2"s,  "videodb://tvshows/titles"s },
+  { "videodb://2/3"s,  "videodb://tvshows/years"s },
+  { "videodb://2/4"s,  "videodb://tvshows/actors"s },
+  { "videodb://2/5"s,  "videodb://tvshows/studios"s },
+  { "videodb://2/9"s,  "videodb://tvshows/tags"s },
+  { "videodb://2s",    "videodb://tvshows"s },
```

Comment:
mistake i wonder if it's worth it, the benefits should be limited to the compiling